clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -853,12 +850,38 @@ protected boolean onJoinPrepare(Timer timer, int generation, String memberId) {
         autoCommitOffsetRequestFuture = null;
         timer.update();
 
-        if (exception != null) {
-            throw new KafkaException("User rebalance callback throws an error", exception);
+        if (exception.isPresent()) {
+            throw new KafkaException("User rebalance callback throws an error", exception.get());
         }
         return true;
     }
 
+    private SortedSet<TopicPartition> eagerPartitionsToRevoke(RebalanceProtocol protocol) {
+        SortedSet<TopicPartition> partitions = new TreeSet<>(COMPARATOR);
+        if (protocol != RebalanceProtocol.EAGER) {
+            return partitions;
+        }
+
+        partitions.addAll(subscriptions.assignedPartitions());
+        return partitions;
+    }
+
+    private void markPendingPartitions() {
```

Comment:
How about `maybeMarkPartitionsPendingRevocation`? Otherwise it's a little unclear what exactly is pending.